### PR TITLE
Rr 29 edit/PUT an inventory item

### PR DESCRIPTION
**Pull request by speedemon-21**
added the backend response to the PUT/update inquiry to the endpoint. Fixed conflicts

### CODE REVIEW

**Review by Green-Eggs-And-SAM — COMMENTED** [review]
Maybe consider validation for the input. Like use the 'update' function to check and make sure that the none of the fields are left blank for example. attached is some of ben's code from the 'add' function for inventory (this code is in the 'develop' branch which you should also merge RR29 with). lots of similarities because both involve inputting lots of information.

Screenshot 2024-05-24 at 11 43 29 AM

**Comment by Green-Eggs-And-SAM** [comment]
from jira:

Response returns 404 if inventory ID is not found

Response returns 400 if unsuccessful because of missing properties in the request body (so check if an empty string)

Response returns 400 if the warehouse_id value does not exist in the warehouses table

Response returns 400 if the quantity is not a number (use the method isNaN() to validate if the input is not a number. so if i try and enter 'dog' for the warehouse id and not a number, it needs to check that and respond with error 400)

Response returns 200 if successful

**Comment by speedemon-21** [comment]
> from jira:
> 
> Response returns 404 if inventory ID is not found
> 
> Response returns 400 if unsuccessful because of missing properties in the request body (so check if an empty string)
> 
> Response returns 400 if the warehouse_id value does not exist in the warehouses table
> 
> Response returns 400 if the quantity is not a number (use the method isNaN() to validate if the input is not a number. so if i try and enter 'dog' for the warehouse id and not a number, it needs to check that and respond with error 400)
> 
> Response returns 200 if successful

Will take a improvement on that. thx

**Comment by Green-Eggs-And-SAM** [comment]
Seems to be an error. i keep trying to update id number 1 with a new quantity but it keeps updating id number 5

Screenshot 2024-05-24 at 12 07 21 PM

**Comment by nicmart-dev** [comment]
Actually seems good as you are updating item id 5, but changing to link to warehouse id 1

**Comment by speedemon-21** [comment]
> Seems to be an error. i keep trying to update id number 1 with a new quantity but it keeps updating id number 5
> 
> Screenshot 2024-05-24 at 12 07 21 PM

Hi. I think you are putting in the endpoint with a specific item id as an index. It won't handle the id update request maybe that's why you ended up with #5.
截屏2024-05-24 下午12 12 33

**Comment by Green-Eggs-And-SAM** [comment]
yes! you're right my apologies. works great now! forgot about the url. great catch.

File: routes/inventory-route.js [hunk 1]
```diff
@@ -9,4 +9,6 @@ router.route("/").get(inventoryController.index);
 
 router.route("/:id").get(inventoryController.findOne); // get single inventory item
 
+router.route("/:id").put(inventoryController.update); // Add the PUT route
```
Comment (nicmart-dev) [inline routes/inventory-route.js:14]: You can actually chain this right in line 10 after the get method

`router.route("/:id").get(inventoryController.findOne).put(inventoryController.update);`

**Review by nicmart-dev — APPROVED** [review]

**Comment by nicmart-dev** [comment]
Looks great to me, just added a very minor note to chain the controller endpoints.

**Pull request merged**